### DISCUSSION TITLE
Getting the Assembly that will be produced by a .NET project...

### DISCUSSION THREAD

**Discussion by DynConcepts** [3F/MvsSln#45] [Q/A]
PRetty sure I am missing, but I have spend a few hours, and just don't see it as a property of the Project type classes.

background: I am processing a huge system. Assembly references rather than project references are used. I need to generate the association between the proj file that creates and assembly and the proj file(s) that reference it...

Thanks in advance.

**Comment by 3F** [comment]
https://github.com/3F/MvsSln/blob/9ce5885282d65ee32bfc03808c83c1113c1928e3/MvsSln/Core/IXProject.cs#L356

And everything related to its metadata if you need

https://github.com/3F/MvsSln/blob/9ce5885282d65ee32bfc03808c83c1113c1928e3/MvsSln/Projects/Item.cs#L61

You can also access to AsmData helper
https://github.com/3F/MvsSln/blob/9ce5885282d65ee32bfc03808c83c1113c1928e3/MvsSln/Projects/Item.cs#L91-L97

If you need a type of the .NET projects,

In [2.5.3 ](https://github.com/3F/MvsSln/releases/tag/2.5.3) you may note the following:

* NEW: Added the following extensions for ProjectItem:

    * IsCs() - Checking both legacy `ProjectType.Cs` and modern `ProjectType.CsSdk` types.
    * IsFs() - Checking both legacy `ProjectType.Fs` and modern `ProjectType.FsSdk` types.
    * IsVb() - Checking both legacy `ProjectType.Vb` and modern `ProjectType.VbSdk` types.
    * **IsSdk()** - While ProjectType cannot inform the actual use of the modern Sdk style in projects,
                current method **will try to detect this** by using the extended logic:
                https://github.com/dotnet/project-system/blob/master/docs/opening-with-new-project-system.md

**Comment by DynConcepts** [comment]
Thanks... I must be being dense.... The AsmData (Assembly) appears to be on an Item type...

What I am looking for (ideally in a project type invariant manner (i.e. the details are encapsulated) is related to:

![image](https://user-images.githubusercontent.com/9969889/102550583-1103e880-408c-11eb-9e21-2d14a9679d66.png)

**Comment by 3F** [comment]
This is called property in the context of msbuild. You just need access to the `AssemblyName` property. Easy to use: https://github.com/3F/MvsSln/blob/9ce5885282d65ee32bfc03808c83c1113c1928e3/MvsSln/Core/IXProject.cs#L180-L183

Note the 2 different values in [PropertyItem](https://github.com/3F/MvsSln/blob/master/MvsSln/Projects/PropertyItem.cs#L35):

* `unevaluatedValue`- raw data, for example: 'Asm.$(edition).$(product)' etc.
* `evaluatedValue` - final data, for example: 'Asm.rel.svc' etc.

The evaluation of the nodes is recursive (ie. each property can contain other properties, and so on); this is part of the msbuild (MvsSln 2.x). But even for 3.0 with IeXod it should follow the same rules.

```csharp
// ~
xp.GetProperty("AssemblyName").evaluatedValue == "VSDependencies"
```

**Comment by 3F** [comment]
hmm,

```csharp
/// <returns>null if no property of that name and scope exists.</returns> 
```
looks like a typo o_o

**Comment by DynConcepts** [comment]
Thanks that was what I could not find. Since it is surfaced so hig in the VS UI, and also a "key group" my mind sort of expected a dedicated Property... Thanks again.
